frontend/src/libs/components/survey-category/survey-category.tsx

```diff
@@ -0,0 +1,45 @@
+import { useCallback } from '#libs/hooks/hooks.js';
+import {
+  type FormControllerRenderProps,
+  type FormFieldValues,
+  type FormPath,
+} from '#libs/types/types.js';
+
+import styles from './styles.module.scss';
+
+type Properties<T extends FormFieldValues> = {
+  field: FormControllerRenderProps<T, FormPath<T>>;
+  label: string;
+};
+
+const SurveyCategory = <T extends FormFieldValues>({
+  field,
+  label,
+}: Properties<T>): JSX.Element => {
+  const { onChange, ...fieldProperties } = field;
+
+  const handleChange = useCallback(() => {
+    let options = field.value as string[];
+    if (options.includes(label)) {
+      options = options.filter((option: string) => option !== label);
+    } else {
+      options.push(label);
+    }
+
+    onChange(options);
+  }, [field, label, onChange]);
```

Comment:
move this outside and pass onPress without requiring additional params
```suggestion
```

Reply:
Done